rmw_implementation/src/functions.cpp

```diff
@@ -61,17 +72,9 @@ get_library()
         ("failed to find shared library of rmw implementation. Searched " + env_var).c_str());
       return nullptr;
     }
-    try {
-      lib = new Poco::SharedLibrary(library_path);
-    } catch (Poco::LibraryLoadException & e) {
-      RMW_SET_ERROR_MSG(
-        ("failed to load shared library of rmw implementation. Exception: " +
-        e.displayText()).c_str());
-      return nullptr;
```

Comment:
See this version of the dynamic loading actually catches exceptions and uses error states properly and returns error codes, making it very C-like. Perhaps the other similar functions in `rosidl_typesupport_c/cpp` should be updated to look and act like this one...